src/Bicep.Core.Samples/Files/InvalidVariables_LF/Completions/objectVarTopLevel.json

```diff
@@ -1,128 +1 @@
-[
-  {
-    "label": "fifth",
-    "kind": "property",
-    "detail": "fifth",
-    "documentation": {
-      "kind": "markdown",
-      "value": "Type: `array`  \n"
-    },
-    "deprecated": false,
-    "preselect": false,
-    "sortText": "2_fifth",
-    "insertTextFormat": "plainText",
-    "insertTextMode": "adjustIndentation",
-    "textEdit": {
-      "range": {},
-      "newText": "fifth"
-    },
-    "commitCharacters": [
-      "."
-    ]
-  },
-  {
-    "label": "first",
-    "kind": "property",
-    "detail": "first",
-    "documentation": {
-      "kind": "markdown",
-      "value": "Type: `bool`  \n"
-    },
-    "deprecated": false,
-    "preselect": false,
-    "sortText": "2_first",
-    "insertTextFormat": "plainText",
-    "insertTextMode": "adjustIndentation",
-    "textEdit": {
-      "range": {},
-      "newText": "first"
-    },
-    "commitCharacters": [
-      "."
-    ]
-  },
-  {
-    "label": "fourth",
-    "kind": "property",
-    "detail": "fourth",
-    "documentation": {
-      "kind": "markdown",
-      "value": "Type: `'test'`  \n"
-    },
-    "deprecated": false,
-    "preselect": false,
-    "sortText": "2_fourth",
-    "insertTextFormat": "plainText",
-    "insertTextMode": "adjustIndentation",
-    "textEdit": {
-      "range": {},
-      "newText": "fourth"
-    },
-    "commitCharacters": [
-      "."
-    ]
-  },
-  {
-    "label": "second",
-    "kind": "property",
-    "detail": "second",
-    "documentation": {
-      "kind": "markdown",
-      "value": "Type: `bool`  \n"
-    },
-    "deprecated": false,
-    "preselect": false,
-    "sortText": "2_second",
-    "insertTextFormat": "plainText",
-    "insertTextMode": "adjustIndentation",
-    "textEdit": {
-      "range": {},
-      "newText": "second"
-    },
-    "commitCharacters": [
-      "."
-    ]
-  },
-  {
-    "label": "sixth",
-    "kind": "property",
-    "detail": "sixth",
-    "documentation": {
-      "kind": "markdown",
-      "value": "Type: `object[]`  \n"
-    },
-    "deprecated": false,
-    "preselect": false,
-    "sortText": "2_sixth",
-    "insertTextFormat": "plainText",
-    "insertTextMode": "adjustIndentation",
-    "textEdit": {
-      "range": {},
-      "newText": "sixth"
-    },
-    "commitCharacters": [
-      "."
-    ]
-  },
-  {
-    "label": "third",
-    "kind": "property",
-    "detail": "third",
-    "documentation": {
-      "kind": "markdown",
-      "value": "Type: `int`  \n"
-    },
-    "deprecated": false,
-    "preselect": false,
-    "sortText": "2_third",
-    "insertTextFormat": "plainText",
-    "insertTextMode": "adjustIndentation",
-    "textEdit": {
-      "range": {},
-      "newText": "third"
-    },
-    "commitCharacters": [
-      "."
-    ]
-  }
-]
\ No newline at end of file
+[]
```

Comment:
This seems odd. Any idea why the completion items are removed?

Reply:
Good catch. I think because of https://github.com/Azure/bicep/pull/1915#discussion_r610022836. I may need to revise that approach...

Reply:
Yeah...

Reply:
https://github.com/Azure/bicep/pull/1915/commits/24402525cd45478abfa8f49adec792fbbcb240d4 - here's is a more targeted fix I came up with to avoid the infinite recursion. I don't love it because it feels very special-case.

I did this because:
* language server functionality depends on the declared type for variable declarations being resolved (rather than just `any`)
* we otherwise have infinite recursion (variable declared type -> variable value assigned type -> for loop assigned type -> object assigned type -> object declared type -> for loop declared type -> variable declared type)

Reply:
Personally I prefer
> removing the language server's dependency on declared type

to be the ultimate fix because that avoid special cases, but within the scope of this PR the fix looks good to me.

Reply:
Also I feel like this can be a question for Anders.